W-11862525-first-draft-for-beta-site-build-CP (#114)
mulesoft/docs-apikit: modules/ROOT/pages/apikit-4-for-rest.adoc

= APIkit for REST
ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]
:page-aliases: apikit-4-generate-workflow.adoc

> APIkit for REST is a framework designed to simplify the development and integration of RESTful APIs. REST (Representational State Transfer) is an architectural style for building web services. With APIkit for REST, you get a set of tools and features to streamline REST API development while ensuring compliance with industry standards.

Reviewer: ```suggestion
Use APIkit for REST to streamline REST API development while ensuring compliance with industry standards. APIkit for REST is a framework designed to simplify the development and integration of RESTful APIs. REST (Representational State Transfer) is an architectural style for building web services. 
```
Small optional suggestion to focus on what APIkit accomplishes

Author: I like it! Thanks!

> APIkit for REST fully supports RAML (RESTful API Modeling Language) and OAS (OpenAPI Specification), which are widely adopted formats for describing RESTful APIs. By leveraging RAML or OAS, you can define your API contracts, enabling consistent design, enhanced collaboration, and clear documentation.

Reviewer: ```suggestion
APIkit for REST fully supports RAML (RESTful API Modeling Language) and OAS (OpenAPI Specification), which are widely adopted formats for describing RESTful APIs. Define your API contracts using RAML or OAS to benefit from consistent design, enhanced collaboration, and clear documentation.
```
Another optional suggestion

Author: Simpler is better! Thanks!

== Features

> * *Automatic Validation*: APIkit for REST automatically validates incoming requests against your API contracts defined in RAML or OAS. This validation ensures that requests adhere to the specified rules and constraints of the API.

Reviewer 2: Kim always suggested using a + to get a horizontal space between the field/ feature and the explanation text, unless the explanation text is very short. This makes it easier to see the information.

* Automatic validation (no bolding or title case)
+
APIkit automatically validates incoming requests against your API contracts defined in RAML or OAS. This validation ensures that requests adhere to the specified rules and constraints of the API.

Author: OK! Thanks!

* *Code Generation*: The framework provides code generation capabilities that help streamline the implementation of your RESTful APIs. APIkit for REST generates implementation templates based on your API contracts.

* *Intelligent Routing*: With APIkit for REST, incoming requests are intelligently routed to the appropriate endpoints based on the defined API contracts. 

> APIkit for REST is a tool for efficient REST API development and integration. By leveraging its features and adhering to industry standards, you can accelerate your API development lifecycle, improve collaboration, and deliver high-quality RESTful APIs.

Reviewer: ```suggestion
```
This sentence is redundant with the intro. If there's new info here, please move it to the intro section and leave only the bulleted lists of features.

Author: Actually, I tried to make a closing paragraph. If it does not work, I may delete it. Would it work well?

Author: No need to address this change as per the editor's comment.
